src/workerd/api/global-scope.h

```diff
@@ -44,7 +44,9 @@ class Navigator: public jsg::Object {
 public:
   kj::StringPtr getUserAgent() { return "Cloudflare-Workers"_kj; }
 #ifdef WORKERD_EXPERIMENTAL_ENABLE_WEBGPU
-  jsg::Ref<api::gpu::GPU> getGPU() { return jsg::alloc<api::gpu::GPU>(); }
+  jsg::Ref<api::gpu::GPU> getGPU(CompatibilityFlags::Reader flags) {
+    return jsg::alloc<api::gpu::GPU>(flags);
```

Comment:
I'm generally not a big fan of constructors that throw like this. Often it can't be avoided but in this case I'd prefer that the `getActor()` and config flag check be performed here rather than in the constructor.